lib/cylc/gui/updater_dot.py

```diff
@@ -303,9 +304,39 @@ def _get_expanded_rows(self):
                 self.expanded_rows.append(self.led_treestore.get_value(
                     self.led_treestore.get_iter(row), 0))
 
+    def _get_selected_rows(self):
+        """Make a note of currently selected rows.
+
+        Populates self.selected_rows with the value of the first column of all
+        selected rows.
+
+        """
+        self.selected_rows = []
+        _, selected_paths = self.led_treeview.get_selection(
+        ).get_selected_rows()
+        model = self.led_treeview.get_model()
+        # Do this with a selected_foreach?
+        for path in selected_paths:
+            self.selected_rows.append(model.get_value(model.get_iter(path), 0))
+
+    def _set_selected_rows(self):
+        """Re-Selects previously selected rows where possible.
+
+        Uses self.selected_rows to determine which rows to select.
+        Warning: This method has not been tested with multiple selection.
+
+        """
+        def should_select_row(model, path, iter_, selection):
```

Comment:
The logic doesn't really make sense within DotUpdater so I encapsulated it (pep227). I could convert it to a lambda if you prefer.